pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -158,6 +157,25 @@ func (c *Repair) runOnce() error {
 		return fmt.Errorf("unable to refresh the service IP block: %v", err)
 	}
 
+	// before we go through snapshots, we fix "PreferDualStack" services
+	// if needed
+	if err = c.fixPreferDualStackServices(); err != nil {
+		return err
```

Comment:
err or event?

Reply:
The func itself now does not return error except in total failure (e.g. listing or generating patch bytes) otherwise it broadcasts events.